satellite/metainfo/metainfo.go

```diff
@@ -470,23 +475,44 @@ func (endpoint *Endpoint) filterValidPieces(ctx context.Context, pointer *pb.Poi
 			return err
 		}
 
-		var remotePieces []*pb.RemotePiece
-		allSizesValid := true
-		lastPieceSize := int64(0)
-		for _, piece := range remote.RemotePieces {
+		type invalidPiece struct {
+			NodeID   storj.NodeID
+			PieceNum int32
+			Reason   string
+		}
 
+		var (
+			remotePieces  []*pb.RemotePiece
+			invalidPieces []invalidPiece
+			lastPieceSize int64
+			allSizesValid = true
+		)
+		for _, piece := range remote.RemotePieces {
 			// Verify storagenode signature on piecehash
 			peerID, ok := peerIDMap[piece.NodeId]
 			if !ok {
-				endpoint.log.Warn("Identity chain unknown for node", zap.String("nodeID", piece.NodeId.String()))
+				endpoint.log.Warn("Identity chain unknown for node. Piece removed from pointer",
+					zap.String("nodeID", piece.NodeId.String()),
+					zap.Int32("pieceID", piece.PieceNum),
+				)
+
+				invalidPieces = append(invalidPieces, invalidPiece{
+					NodeID:   piece.NodeId,
+					PieceNum: piece.PieceNum,
+					Reason:   "Identity chain unknown for node",
+				})
 				continue
 			}
 			signee := signing.SigneeFromPeerIdentity(peerID)
 
 			err = endpoint.validatePieceHash(ctx, piece, limits, signee)
 			if err != nil {
-				// TODO maybe this should be logged also to uplink too
-				endpoint.log.Warn("Problem validating piece hash", zap.Error(err))
+				endpoint.log.Warn("Problem validating piece hash. Pieces removed from pointer", zap.Error(err))
```

Comment:
Don't we need to log the Node ID and Piece ID here like in the previous log? Or is this info already in the error message?

Reply:
It's in the error message, `validatePieceHash` adds them https://github.com/storj/storj/blob/aba9a9028f9f70ce976822827abf492b121d4357/satellite/metainfo/validation.go#L365:27